packages/aws-appsync/src/client.js

```diff
@@ -28,51 +84,47 @@ class AWSAppSyncClient extends ApolloClient {
 
     /**
      *
-     * @param {string} url
+     * @param {object} appSyncOptions
      * @param {ApolloClientOptions<InMemoryCache>} options
      */
-    constructor({ url, region, auth, conflictResolver, complexObjectsCredentials, disableOffline = false }, options) {
-        if (!url || !region || !auth) {
+    constructor({
+        url,
+        region,
+        auth,
+        conflictResolver,
+        complexObjectsCredentials,
+        cacheOptions,
+        disableOffline = false
+    } = {}, options = {}) {
+        const { cache: customCache, link: customLink } = options;
+
+        if (!customLink && (!url || !region || !auth)) {
```

Comment:
I would add on the error message **or customLink**

Reply:
Sounds good!